packages/keystone/runtimeStats.js

```diff
@@ -0,0 +1,171 @@
+                    runtimeStats: { activeRequestsIds: Array.from(activeRequestsIds.keys()), ...otherStats },
+                })
+
+                metrics.gauge({ name: 'runtimeStats.requestsCount.total', value: runtimeStats.totalRequestsCount })
+
+                for (const type of Object.keys(runtimeStats.totalRequestsCountByType)) {
+                    metrics.gauge({
+                        name: `runtimeStats.requestsCount.type.${camelCase(type)}.total`,
+                        value: runtimeStats.totalRequestsCountByType[type],
+                    })
+                }
+
+                for (const target of Object.keys(runtimeStats.totalRequestsCountByTarget)) {
+                    metrics.gauge({
+                        name: `runtimeStats.requestsCount.target.${camelCase(target)}.total`,
+                        value: runtimeStats.totalRequestsCountByTarget[target],
+                    })
+                }
+            }, 1000)
+        }
+
+        // nosemgrep: javascript.express.security.audit.express-check-csurf-middleware-usage.express-check-csurf-middleware-usage
+        const app = express()
+
+        app.use(function runtimeStatsMiddleware (req, res, next) {
+            try {
+                const url = new URL(`${conf['SERVER_URL']}${req.url}`).pathname
+                const method = (req.method || 'get').toLowerCase()
+
+                const possibleXTargetValues = (conf[X_TARGET_OPTIONS_VAR_NAME] || '').split(',').filter(Boolean)
+                const xTargetHeader = req.headers['x-target']
+
+                const xTarget = possibleXTargetValues.includes(xTargetHeader) ? xTargetHeader : 'other'
+
+                const isPost = method === 'post'
+
+                const isAdminApiUrl = url === '/admin/api'
+                const isApiUrl = url.startsWith('/api')
+                const isOidcUrl = url.startsWith('/oidc')
+                const isWellKnownUrl = url.startsWith('/.well-known')
+                const isHealthCheckUrl = url.startsWith(DEFAULT_HEALTHCHECK_URL)
+
+                /** @type {'api' | 'graphql' | 'oidc' | 'wellKnown' | 'healthCheck' | 'other'} */
+                let requestType = 'other'
+
+                if (isPost && isAdminApiUrl) {
+                    requestType = 'graphql'
+                } else if (isApiUrl) {
+                    requestType = 'api'
+                } else if (isOidcUrl) {
+                    requestType = 'oidc'
+                } else if (isWellKnownUrl) {
+                    requestType = 'wellKnown'
+                } else if (isHealthCheckUrl) {
+                    requestType = 'healthCheck'
+                }
+
+                runtimeStats.activeRequestsIds.add(req.id)
+
+                runtimeStats.activeRequestsCountByType[requestType] = (runtimeStats.activeRequestsCountByType[requestType] || 0) + 1
+                runtimeStats.activeRequestsCountByTarget[xTarget] = (runtimeStats.activeRequestsCountByTarget[xTarget] || 0) + 1
```

Comment:
Not a mandatory, but we can also init all targets and types during constructor.
It might be helpful for 2 reasons:
- Each pod will have metrics of each type === easier analytics (dont need to deal with "No data" in grafana and others)
- This code becomes simplier by `runtimeStats.activeRequestsCountByType[requestType]++`